game/ato/flightplans/capbuilder.py

```diff
@@ -34,8 +38,43 @@ def cap_racetrack_for_objective(
             if airfield.captured != self.is_player:
                 closest_airfield = airfield
                 break
+            elif closest_friendly_field is None:
+                closest_friendly_field = airfield
         else:
-            raise PlanningError("Could not find any enemy airfields")
+            if barcap:
+                # If planning a BARCAP, we should be able to find at least one enemy
+                # airfield. If we can't, it's an error.
+                raise PlanningError("Could not find any enemy airfields")
+            else:
+                # if we cannot find any friendly or enemy airfields other than the target,
+                # there's nothing we can do
+                if closest_friendly_field is None:
+                    raise PlanningError(
+                        "Could not find any enemy or friendly airfields"
+                    )
+
+                # If planning other race tracks (TARCAPs, currently), the target may be
+                # the only enemy airfield. In this case, set the race track orientation using
+                # a virtual point equi-distant from but opposite to the target from the closest
+                # friendly airfield like below, where F is the closest friendly airfield, T is
+                # the sole enemy airfield and V the virtual point
+                #
+                # F ---- T ----- V
+                #
+                # We need to create this virtual point, rather than using F to make sure
+                # the race track is aligned towards the target.
+                closest_friendly_field_position = copy.deepcopy(
+                    closest_friendly_field.position
+                )
+                closest_airfield = closest_friendly_field
+                closest_airfield.position.x = (
```

Comment:
Good enough so will merge as-is, but iirc you can do `(point_a + point_b) / 2` and avoid the copy. Maybe I'm confusing shapely and pydcs, but I thought all the basic vector math worked.